### ISSUE TITLE
Bot to automatically update a transcludeable template for ACC Status

### ISSUE THREAD

**Issue by Dane2007** [enwikipedia-acc/waca#437]
Create or implement functionality into an existing bot to update a template on-wiki that ACC users can add which indicates the status of ACC depending on the number of requests.  This may be helpful in stimulating more activity by tool users to actively handle requests at the ACC system.

I am planning on researching/starting work on this later this month.  Any comments or suggestions are appreciated.

**Comment by stwalkerster** [comment]
If you're wanting to do this as a new bot, I advise doing this in a separate repository to this one - we can easily create one under the github organisation if you want - that way the actual tool code and this are kept separate. From a technical standpoint, I've got some JS code on-wiki which adds this to my toolbox:

```js
if (typeof(userACC) === 'undefined'){ var userACC = mw.config.get( 'wgUserName' ).replace(" ", "_"); }
$.ajax({
	url: '//accounts.wmflabs.org/api.php?action=status',
	dataType: 'xml',
	success: function(ACCstatus) {
		var requests = parseInt($(ACCstatus).find('status').attr('open'), 10);

			mw.util.addPortletLink('p-tb', '//accounts.wmflabs.org/acc.php?tplUsername=' + userACC, requests + ' requested accounts!', 'p-acc', 'Go to the Account Creation Interface');
		
	}
});
```
which looks like this:
![image](https://user-images.githubusercontent.com/722710/26945036-6994084c-4c82-11e7-8b13-2380f2ae7c88.png)

You should be able to use the tool API in the same way.

Remember if you want to do this as an actual bot, you'll probably need a [bot approval](http://enwp.org/WP:BRFA)

**Comment by Dane2007** [comment]
Thanks.  I'll do this work in another repository if necessary but i'll leave the issue here for now for ease of access/comments.  In terms of your example code, is that JS just in your themes JS file?  That code looks useful in doing this.

**Comment by stwalkerster** [comment]
Yeah, it is, [it's here](enwp.org/User:stwalkerster/common.js)

**Comment by Dane2007** [comment]
@stwalkerster The API call for the number of accounts, is this the same in newinternal?  Reviewing some code for this project.

**Comment by stwalkerster** [comment]
@Dane2007 Yep, I'm aware of a few different things using this for status requests etc, so I've made an effort not to break those

**Comment by dqwiki** [comment]
While this definitely seems innocent, please add me as a reviewer to the pull request when it happens, as anything going outside the tool needs to be well protected and affects our privacy policies. I will also need an IRC ping when it happens.

**Comment by stwalkerster** [comment]
@dqwiki I'd be very surprised if a pull request comes out of this - this is the wrong place for this issue to be open as it's not actually going to change the tool code, but instead use the existing tool API. I also find it hard to see how this would have any privacy policy impact at all given the nature of this data.

**Comment by Dane2007** [comment]
I’m working on this on my own repo as this isn’t directly ACC code. This is grabbing the data that’s already accessible from the API and simply updating the number of requests and the message for the template. However I will ping you @dqwiki so you can see this code prior to taking it for bot approval/trial.

**Comment by dqwiki** [comment]
Oh ok, if were using the existing API, without changes, that's fine then. My thought was that there were API modifications going on to make this functionable.
